test/utils/browser.js

```diff
@@ -30,32 +32,86 @@ exports.openMusic = function (options) {
+      if (arg === '40000') {
+        this.browser = wd.remote('hub.browserstack.com', 80);
+      }
+    }.bind(this));
     global.browser = this.browser;
   });
   before(function openBrowser (done) {
-    this.browser.init({browserName: 'chrome'}, done);
+    var that = this;
+    this.browser.init({
+      browserName: 'firefox',
+      name: 'Google Music Core Test Suite',
+      project: 'Google Music Core Test Suite',
+      platform: 'WIN8',
+      'browserstack.user': BS_USER,
+      'browserstack.key': BS_KEY
+    }, function () {
+      that.browser.maximize();
+      that.browser.setAsyncScriptTimeout(30000, done);
+    });
   });
   before(function navigateToMusicBeforeLogin (done) {
     this.browser.get(url, done);
   });
   before(function handleLoginViaCookies (done) {
     var browser = this.browser;
-    async.forEach(cookies, function setCookies (cookie, cb) {
-      // If the cookie is not for .google.com, skip it
-      // DEV: As discovered by Burp suite's repeater, we only need `SID`, `HSID`, `SSID` but this is simpler
-      if (cookie.domain !== '.google.com') {
-        process.nextTick(cb);
-      // Otherwise, set it
-      } else {
-        browser.setCookie(cookie, cb);
-      }
-    }, done);
+    browser.elementByCssSelector('[data-action=signin]', function (err, el) {
+      if (err) { done(err); }
+
+      el.click(function (err) {
+        if (err) { done(err); }
+
+        browser.elementById('Email', function (err, el) {
+          if (err) { done(err); }
+
+          var emailSetCondition = 'document.getElementById("Email").value == "' + username + '"';
+          browser.waitForConditionInBrowser('document.getElementById("Email") != null', function () {
+            browser.execute('document.getElementById("Email").value = "' + username + '"');
+            browser.waitForConditionInBrowser(emailSetCondition, function (err) {
+              browser.elementById('next', function (err, el) {
+                if (err) { done(err); }
+
+                el.click(function (err) {
+                  if (err) { done(err); }
+
+                  browser.waitForConditionInBrowser('document.getElementById("Passwd") != null', function () {
+                    var passwordSetCondition = 'document.getElementById("Passwd").value == \'' + password + '\'';
+
+                    browser.execute('document.getElementById("Passwd").value = "' + password + '"');
+                    browser.waitForConditionInBrowser(passwordSetCondition, function () {
+                      browser.elementById('signIn', function (err, el) {
+                        el.click(function (err) {
+                          if (err) { done(err); }
+
+                          var doneLoginCondition =
+                            'window.location.href.indexOf("https://play.google.com/music") === 0';
+                          browser.waitForConditionInBrowser(doneLoginCondition, function () { done(); });
+                        });
```

Comment:
This pyramid of callbacks is absurd. Please use something like the `async` module to avoid it.

```js
async.waterfall([
  function findEmailEl (cb) {
    browser.elementById('Email', cb);
  },
  function inputEmail (el, cb) {
    browserUtils.execute(/* code to input email ... */
  }
], done);
```